### ISSUE TITLE
Update Developer Documentation

### ISSUE THREAD

**Issue by stevejgordon** [HTBox/allReady#2011]
After the VS 2017/csproj update https://github.com/HTBox/allReady/blob/master/docs/prerequisite_install_guide/prerequisite_install_guide.md is now out of date and misleading for new contributors.

We need to review this and ensure it reflects the current on-boarding/setup process for devs.

**Comment by marigerr** [comment]
If this is open to new contributors, I can help with this.   I recently got everything installed with VS 2017, so familiar with some of the needed changes to documentation.

**Comment by stevejgordon** [comment]
@marigerr Yes, please feel free to jump in on this. Would be greatly appreciated as you have the most recent experience.

**Comment by marigerr** [comment]
great!  I'll get started.

**Comment by jamesatgreymetis** [comment]
I worked through the guide yesterday and needed to install js tools from scratch, if it helps here is a snapshot of the current version numbers once I was finished:

git version 2.13.1.windows.1
node 6.11.0
npm 3.10.10
bower 1.8.0
gulp CLI version 1.3.0

The step to upgrade npm to v3 was not needed as it was already v3.10.10

Installing bower returned this message:
npm WARN deprecated bower@1.8.0: ..psst! While Bower is maintained, we recommend Yarn and Webpack for new front-end projects! Yarn's advantage is security and reliability, and Webpack's is support for both CommonJS and AMD projects. Currently there's no migration path but we hope you'll help us figure out one.

**Comment by marigerr** [comment]
Thanks @jamesatgreymetis   I haven't had any problem myself running the latest versions of everything, so I was going to assume that it's ok to include latest versions of all the tools in docs?

I'm planning to do a new install of everything on a second machine and get screenshots for both VS 2017 installs and latest versions of node, git installs etc.  

Unsure if they are planning to switch from bower to webpack (or other tool), but might be useful for timebeing to include something in documentation that can ignore the warning.

**Comment by MisterJames** [comment]
Closed via #2014

**Comment by iloabn** [comment]
Last comment says that this is closed, but it still seems open.
I tried to install the project yesterday and noticed a slight difference between the short and longer installation instructions as well as a missing step.

The [short](https://github.com/HTBox/allReady/wiki/Developer-Setup) instructions doesn't mention .NET Core 2 SDK or the `F# language support`.

The [longer](https://github.com/HTBox/allReady/blob/master/docs/prerequisite_install_guide/prerequisite_install_guide.md) mentions `F# language support` but not .NET Core 2 SDK.

I'd love to include some light text on how to install it in the prerequisite guide and a list item in the shorter one. (but I guess I can't edit that since it's in the Wiki?)

**Comment by iloabn** [comment]
@stevejgordon What more work is required on this before you can consider this issue closed? 🙂

**Comment by jonparker** [comment]
Hi @iloabn  & @stevejgordon. Just an FYI, @SteveKennedy has upgraded the Mobile project as part of #2031  so these instructions will need to be updated once this gets merged.

**Comment by c0g1t8** [comment]
Anyone working on this?  If not, I'd like to _refactor_ the developer documentation.  Some of the information is out of date and need to be re-validated.

**Comment by MisterJames** [comment]
@c0g1t8 This one is yours to pick up if you like!

It _was_ to be closed, but tools continue to evolve and updates are likely required from time to time. ...and this could be a time!
